src/VCR/MatchObserver.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+namespace VCR;
+
+class MatchObserver
```

Comment:
You're correct that it's a service. I don't like that. I will look into making it a value object.

I agree with you on the name. I originally was thinking that the observer pattern would apply here, and kept that name. I like your name.